### PR TITLE
Switching the Showcase to the new backend system & 1.23.0-next.3 pre-release

### PR DESCRIPTION
**Pull request by davidfestal**
## Description

This PR is a proof PR of everything involved in switching the Showcase application to the new backend system, and bumping it at the same time to the 1.23.0-next.3 pre-release, in order to prepare for the adoption of the upcoming 1.23.0 release.

## Which issue(s) does this PR fix

This PR is related to EPIC https://github.com/janus-idp/backstage-showcase/issues/834

## PR acceptance criteria

Please make sure that the following steps are complete:

- [ ] GitHub Actions are completed and successful
- [ ] Unit Tests are updated and passing
- [ ] E2E Tests are updated and passing
- [ ] Documentation is updated if necessary (requirement for new features)
- [ ] Add a screenshot if the change is UX/UI related

### CODE REVIEW

**Comment by davidfestal** [comment]
> Took a moment to give this a try locally and it seems to be working nicely with my basic config at least, nice!
> 
> I had a peek at the browser console and noticed these warnings, I think we need to be mindful of these as well, as this indicates a mismatch between what a plugin is expecting and what module is being provided by the application shell.
> 
> image
> 
> At the very least, if we do find some UI issues this would be the first warning to solve 😄

Thanks very much @gashcrumb for your testing.
I'll have a look at it.

**Comment by AndrienkoAleksandr** [comment]
@davidfestal I test rbac-backend plugin for this pr using helm. I tested it with scaffolder, catalog and jenkins plugins - it works fine. But when I tried to enable rbac frontend plugin -  then init container failed. I used default rbac configuration: 

```
      - package: ./dynamic-plugins/dist/janus-idp-plugin-rbac
        disabled: false
        pluginConfig:
          dynamicPlugins:
            frontend:
              janus-idp.backstage-plugin-rbac:
                mountPoints:
                  - mountPoint: admin.page.rbac/cards
                    module: RbacPlugin
                    importName: RbacPage
                    config:
                      layout:
                        gridColumn: "1 / -1"
                        width: 100vw
                      props:
                        useHeader: false
                dynamicRoutes:
                  - path: /admin/rbac
                    module: RbacPlugin
                    importName: RbacPage
```

Event error: Back-off restarting failed container

**Comment by gashcrumb** [comment]
>  - package: ./dynamic-plugins/dist/janus-idp-plugin-rbac

Should this be `janus-idp-backstage-plugin-rbac` instead?

**Comment by AndrienkoAleksandr** [comment]
@gashcrumb, thanks you are right. rbac ui works too, but with fixed path.

**Comment by gashcrumb** [comment]
@davidfestal created #972 about the version warning messages in the browser.

File: packages/backend/src/modules/rbacDynamicPluginsModule.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+} from '@backstage/backend-dynamic-feature-service';
+
+const pluginIDProviderServiceRef = createServiceRef<PluginIdProvider>({
+  id: 'pluginIDProvider',
+  scope: 'root',
+});
+
+/*
+// When (if ?) the BackendFeatureRegistrationObserver addition is accected / merged upstream,
+// theh the following commented code would be the right one.
```
Comment (schultzp2020) [inline packages/backend/src/modules/rbacDynamicPluginsModule.ts:20]:
```suggestion
// the the following commented code would be the right one.
```

Link to PR?

**Review by schultzp2020 — CHANGES_REQUESTED** [review]

File: packages/app/package.json [hunk 2]
```diff
@@ -14,28 +14,28 @@
...
-    "@backstage/plugin-catalog-react": "1.9.3",
-    "@backstage/plugin-home": "0.6.1",
-    "@backstage/plugin-org": "0.6.19",
-    "@backstage/plugin-permission-react": "0.4.19",
-    "@backstage/plugin-scaffolder": "1.17.1",
-    "@backstage/plugin-search": "1.4.5",
-    "@backstage/plugin-search-react": "1.7.5",
-    "@backstage/plugin-user-settings": "0.8.0",
-    "@backstage/theme": "0.5.0",
+    "@backstage/app-defaults": "^1.5.0-next.3",
```
[the remark this answers is not in the document]
Comment (davidfestal) [reply packages/app/package.json]: Fixed

File: packages/backend/src/modules/rbacDynamicPluginsModule.ts (hunk 1 above)
Comment (davidfestal) [reply packages/backend/src/modules/rbacDynamicPluginsModule.ts]: Linked has been added

**Review by schultzp2020 — APPROVED** [review]
/approve
/lgtm

**Pull request merged**